packages/flutter_tools/templates/plugin/android-java.tmpl/src/main/java/androidIdentifier/pluginClass.java.tmpl

```diff
@@ -1,5 +1,37 @@
 package {{androidIdentifier}};
 
+{{#useNewAndroidEmbedding}}
+{{#androidX}}
+import androidx.annotation.NonNull;
+{{/androidX}}
+{{^androidX}}
+import android.support.annotation.NonNull;
+{{/androidX}}
+import io.flutter.embedding.engine.plugins.FlutterPlugin;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
+import io.flutter.plugin.common.MethodChannel.Result;
+
+/** {{pluginClass}} */
+public class {{pluginClass}} implements FlutterPlugin, MethodCallHandler {
```

Comment:
Plugins generated with this template will not be backward compatible with apps that are using the v1 embedding.

Should we have a `registerWith` here as well? at least until we are ready to deprecate the v1 embedding?

@blasten @xster @matthew-carroll @mklim

Reply:
so far, the plan is to exit the tool if an app using the v1 embedding attempts to use a plugin using the v2 embedding. Issue: https://github.com/flutter/flutter/issues/43779

Reply:
Offline discussion - we should probably see enough adoption of v2 embedding in Flutter apps and be willing to deprecate it before we make it easy to create plugins that do not support the v1 embedder.

Reply:
I also agree with that conclusion. For at least 2-3 quarters, I would recommend generating backward compatible plugins with appropriate comment instructions.